src/inmanta/server/services/fileservice.py

```diff
@@ -131,13 +131,14 @@ def stat_file_internal(self, files: Iterable[str]) -> List[str]:
         """
         Return which files in the list don't exist on the server
         """
-        response: List[str] = []
+        response: Dict[str, object] = {}
```

Comment:
I decided to use a dict here where only the keys are of interest to have an efficient set-like behavior and preserving insertion order compared to the previous implementation.
I'm not sure whether this is truly necessary or not

Reply:
Looks good to me. Perhaps add a comment.